shuffle/cli/src/account.rs

```diff
@@ -57,75 +52,94 @@ pub fn handle(home: &Home, root: Option<PathBuf>) -> Result<()> {
-    if root.is_some() {
-        home.save_root_key(
-            root.ok_or_else(|| anyhow::anyhow!("Invalid root key path"))?
-                .as_path(),
-        )?;
-    }
-    let mut treasury_account = get_treasury_account(&client, home.get_root_key_path());
-
-    create_local_accounts(home, client, &mut treasury_account)
-}
-
-pub fn create_local_accounts(
-    home: &Home,
-    client: BlockingClient,
-    treasury_account: &mut LocalAccount,
-) -> Result<()> {
     let factory = TransactionFactory::new(ChainId::test());
 
-    home.generate_shuffle_accounts_path()?;
-    home.generate_shuffle_latest_path()?;
+    if let Some(input_root_key) = root {
+        network_home.save_root_key(input_root_key.as_path())?
+    }
 
-    let new_account_key = home.generate_key_file()?;
-    let public_key = new_account_key.public_key();
-    home.generate_address_file(&public_key)?;
-    let new_account = LocalAccount::new(
-        AuthenticationKey::ed25519(&public_key).derived_address(),
-        new_account_key,
-        0,
-    );
+    network_home.generate_network_accounts_path_if_nonexistent()?;
+    network_home.generate_network_latest_account_path_if_nonexistent()?;
 
-    // Create a new account.
-    create_account_onchain(treasury_account, &new_account, &factory, &client)?;
+    let mut treasury_account = get_treasury_account(&client, home.get_root_key_path());
+    let new_account = generate_new_account(&network_home)?;
+    create_account_onchain(&mut treasury_account, &new_account, &factory, &client)?;
 
-    home.generate_shuffle_test_path()?;
-    let test_key = home.generate_testkey_file()?;
-    let public_test_key = test_key.public_key();
-    home.generate_testkey_address_file(&test_key.public_key())?;
-    let test_account = LocalAccount::new(
-        AuthenticationKey::ed25519(&public_test_key).derived_address(),
-        test_key,
-        0,
-    );
+    network_home.generate_shuffle_test_path_if_nonexistent()?;
+    let test_account = generate_test_account(&network_home)?;
+    create_account_onchain(&mut treasury_account, &test_account, &factory, &client)
+}
 
-    create_account_onchain(treasury_account, &test_account, &factory, &client)
+fn check_node_deployed_if_localhost_used(home: &Home, network: &Network) -> Result<()> {
+    match network.get_name().as_str() {
+        LOCALHOST_NAME => match home.get_node_config_path().is_dir() {
+            true => Ok(()),
+            false => Err(anyhow!(
+                "A node hasn't been created yet! Run shuffle node first"
+            )),
+        },
+        _ => Ok(()),
+    }
 }
 
-pub fn confirm_user_decision(home: &Home) -> bool {
-    let key_path = home.get_latest_key_path();
+fn user_wants_another_key(network_home: &NetworkHome) -> bool {
+    let key_path = network_home.get_latest_account_key_path();
     let prev_key = generate_key::load_key(&key_path);
     println!(
         "Private Key already exists: {}",
         ::hex::encode(prev_key.to_bytes())
```

Comment:
We never want to print the private key. Print the public key or derived address.